spec.html

```diff
@@ -32162,26 +32223,122 @@ <h1>
+              <td>a String</td>
+              <td>The primary time zone identifier that [[Identifier]] resolves to.</td>
+            </tr>
+          </table>
+        </emu-table>
+        <emu-note>
+          <p>If [[Identifier]] is a primary time zone identifier, then [[Identifier]] will be [[PrimaryIdentifier]].</p>
+        </emu-note>
+        <emu-note>
+          <p>In time zone aware implementations it is required (and for all other implementations, recommended) that both [[Identifier]] and [[PrimaryIdentifier]] be a Zone or Link name from the IANA Time Zone Database <a href="https://www.iana.org/time-zones/">https://www.iana.org/time-zones/</a>.</p>
+        </emu-note>
+      </emu-clause>
+
+      <emu-clause id="sec-availabletimezoneidentifiers" type="implementation-defined abstract operation">
+        <h1>AvailableTimeZoneIdentifiers ( ): a List of Time Zone Identifier Records</h1>
         <dl class="header">
           <dt>description</dt>
-          <dd>It returns a String value representing the host environment's current time zone, which is either a String representing a UTC offset for which IsTimeZoneOffsetString returns *true*, or a String identifier accepted by GetNamedTimeZoneEpochNanoseconds and GetNamedTimeZoneOffsetNanoseconds.</dd>
+          <dd>
+            Its result describes all available time zone identifiers in this implementation, as well as the primary time zone identifier corresponding to each available identifier.
+            The List is sorted by [[Identifier]] of each Time Zone Identifier Record.
+          </dd>
         </dl>
 
-        <p>An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the DefaultTimeZone abstract operation as specified in the ECMA-402 specification.</p>
-        <p>The default implementation of DefaultTimeZone, to be used for ECMAScript implementations that do not include local political rules for any time zones, performs the following steps when called:</p>
+        <emu-alg>
+          1. Let _identifiers_ be a List of all time zone identifiers that are supported by the implementation.
+          1. NOTE: For an implementation that does not include local political rules for any time zones, _identifiers_ must be a list with only one item: *"UTC"*.
+          1. Assert: No element of _identifiers_ is an ASCII-case-insensitive match for any other element.
+          1. [declared="comparefn"] Sort _identifiers_ in order as if an Array of the same values had been sorted using %Array.prototype.sort% using *undefined* as _comparefn_.
+          1. Let _result_ be a new empty List.
+          1. For each element _identifier_ of _identifiers_, do
+            1. Let _primary_ be _identifier_.
+            1. If _identifier_ is a non-primary time zone identifier and _identifier_ is not *"UTC"*, then
+              1. Set _primary_ to the name of the primary time zone identifier that _identifier_ resolves to.
```

Comment:
What does it mean for a time zone identifier (which is one of a particular set of String values) to "resolve" to something? Do you mean specifically to do resolution specifically according to tzdata's rules for resolving Link names? If so, this should spell that out.

Note that "time zone aware" as currently defined only means that an implementation supports

> local political rules of time zones represented by all other Zone and Link names (and <b>only</b> such names) from the IANA Time Zone Database

That is, the requirement is only that it is aware of the names in tzdata, not that it actually has a full copy of the IANA Time Zone Database lying around with which to do resolution.

Reply:
> > local political rules of time zones represented by all other Zone and Link names (and only such names) from the IANA Time Zone Database
> 
> That is, the requirement is only that it is aware of the names in tzdata, not that it actually has a full copy of the IANA Time Zone Database lying around with which to do resolution.

Good point. I adapted text from elsewhere to make this more thorough: 

> <dfn id="time-zone-aware">Time zone aware</dfn> ECMAScript implementations support *"UTC"* and all other Zone and Link names (and <b>only</b> such names) from the IANA Time Zone Database, the mapping between Link names and corresponding Zone names, and the UTC offsets and transitions associated with Zone names that represent local political rules for that time zone.

Better?

> What does it mean for a time zone identifier (which is one of a particular set of String values) to "resolve" to something? Do you mean specifically to do resolution specifically according to tzdata's rules for resolving Link names? If so, this should spell that out.

👍  I updated to this text: 

> 1. Set _primary_ to the name of the primary time zone identifier that _identifier_ resolves to, according to the rules for resolving Link names in the IANA Time Zone Database.